File: Siddhi mam Tasks/Day 32/Api_Calling/app/src/main/java/com/example/api_calling/Retrofitclient.java
package com.example.api_calling;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class Retrofitclient {

    private static final String BASE_URL = "https://gorest.co.in/";
    private static Retrofit retrofit = null;

    public static Apiinterface getRetrofitClient()
    {
        if(retrofit==null)
        {
            retrofit = new Retrofit.Builder()
                    .baseUrl(BASE_URL).addConverterFactory(GsonConverterFactory.create()).build();
        }

        return retrofit.create(Apiinterface.class);
    }
}

File: Siddhi mam Tasks/Day 42/RetrofitProduct/app/src/main/java/com/example/retrofitproduct/RetrofitClient.kt
package com.example.retrofitproduct

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.create

object RetrofitClient {

    private const val BASE_URL = "https://fakestoreapi.com/"

    val api: APIServices by lazy {
        Retrofit.Builder()
            .baseUrl(BASE_URL)
            .addConverterFactory(GsonConverterFactory.create())
            .build()
            .create(APIServices::class.java)
    }
}


